mm2src/coins/tendermint/tendermint_token.rs

```diff
@@ -0,0 +1,583 @@
+
+    fn sign_message(&self, message: &str) -> SignatureResult<String> { self.platform_coin.sign_message(message) }
+
+    fn verify_message(&self, signature: &str, message: &str, address: &str) -> VerificationResult<bool> {
+        self.platform_coin.verify_message(signature, message, address)
+    }
+
+    fn my_balance(&self) -> BalanceFut<CoinBalance> {
+        let coin = self.clone();
+        let fut = async move {
+            let balance_denom = coin.platform_coin.balance_for_denom(coin.denom.to_string()).await?;
+            Ok(CoinBalance {
+                spendable: big_decimal_from_sat_unsigned(balance_denom, coin.decimals),
+                unspendable: BigDecimal::default(),
+            })
+        };
+        Box::new(fut.boxed().compat())
+    }
+
+    fn base_coin_balance(&self) -> BalanceFut<BigDecimal> { self.platform_coin.my_spendable_balance() }
+
+    fn platform_ticker(&self) -> &str { self.platform_coin.ticker() }
+
+    fn send_raw_tx(&self, tx: &str) -> Box<dyn Future<Item = String, Error = String> + Send> {
+        self.platform_coin.send_raw_tx(tx)
+    }
+
+    fn send_raw_tx_bytes(&self, tx: &[u8]) -> Box<dyn Future<Item = String, Error = String> + Send> {
+        self.platform_coin.send_raw_tx_bytes(tx)
+    }
+
+    fn wait_for_confirmations(
+        &self,
+        tx: &[u8],
+        confirmations: u64,
+        requires_nota: bool,
+        wait_until: u64,
+        check_every: u64,
+    ) -> Box<dyn Future<Item = (), Error = String> + Send> {
+        self.platform_coin
+            .wait_for_confirmations(tx, confirmations, requires_nota, wait_until, check_every)
+    }
+
+    fn wait_for_htlc_tx_spend(
+        &self,
+        transaction: &[u8],
+        secret_hash: &[u8],
+        wait_until: u64,
+        from_block: u64,
+        swap_contract_address: &Option<BytesJson>,
+    ) -> TransactionFut {
+        self.platform_coin.wait_for_htlc_tx_spend(
+            transaction,
+            secret_hash,
+            wait_until,
+            from_block,
+            swap_contract_address,
+        )
+    }
+
+    fn tx_enum_from_bytes(&self, bytes: &[u8]) -> Result<TransactionEnum, MmError<TxMarshalingErr>> {
+        self.platform_coin.tx_enum_from_bytes(bytes)
+    }
+
+    fn current_block(&self) -> Box<dyn Future<Item = u64, Error = String> + Send> { self.platform_coin.current_block() }
+
+    fn display_priv_key(&self) -> Result<String, String> { self.platform_coin.display_priv_key() }
+
+    fn min_tx_amount(&self) -> BigDecimal { big_decimal_from_sat(MIN_TX_SATOSHIS, self.decimals) }
+
+    /// !! This function includes dummy implementation for P.O.C work
+    fn min_trading_vol(&self) -> MmNumber { MmNumber::from("0.00777") }
+}
+
+#[async_trait]
+#[allow(unused_variables)]
+impl MmCoin for TendermintToken {
+    fn is_asset_chain(&self) -> bool { false }
+
+    fn withdraw(&self, req: WithdrawRequest) -> WithdrawFut {
+        let coin = self.platform_coin.clone();
```

Comment:
To make it a bit clearer, could you call it `platform` instead of `coin`, please?